frontend/pages/queries/live/screens/RunQuery.tsx

```diff
@@ -47,12 +51,14 @@ const RunQuery = ({
   const runQueryInterval = useRef<any>(null);
   const globalSocket = useRef<any>(null);
   const previousSocketData = useRef<any>(null);
+  const responseCount = useRef(RESPONSE_COUNT_ZERO);
```

Comment:
I think you are modifying the underlying `RESPONSE_COUNT_ZERO` object here. Is that what you want to do? I think it might be better to copy the object when you initialize the `useRef` instead of passing by reference the const. That way the ref is pointing to a different object.

```ts
const responseCount = useRef({...RESPONSE_COUNT_ZERO});
```

Reply:
great point